### ISSUE TITLE
Connection pooling presumably not working as expected

### ISSUE THREAD

**Issue by fillest** [haproxy/haproxy#65]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 1.9.4-1ppa1~bionic 2019/02/07
Linux qwe 4.15.0-1011-aws #11-Ubuntu SMP Thu Jun 14 13:15:51 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

## What's the configuration?

```
global
	daemon
	nbproc 1
	nbthread 16
	maxconn 100000
defaults
	timeout connect 4s
	timeout client  50s
	timeout server  12s
	balance roundrobin
	http-reuse always
	option http-keep-alive
frontend http-in
	bind :80
	bind :443 ssl crt qwe.pem
	default_backend asd
backend asd
	server server1 1.2.3.4:80 check port 80 inter 1000ms fastinter 1000ms downinter 1000ms rise 2 fall 2 pool-max-conn -1 pool-purge-delay 0
	option httpchk GET / 
```

## Steps to reproduce the behavior

1. I'm using `hey` tool for load testing. It can be set up this way:
    1. install golang e.g. `curl https://storage.googleapis.com/golang/go1.12.1.linux-amd64.tar.gz | tar xzf - -C /tmp/ && sudo mv /tmp/go /usr/local/go`
    2. install hey `go install github.com/rakyll/hey`
2. `~/go/bin/hey -c 700 -q 2 -z 30s -t 10 -disable-keepalive https://myhost/`. It sustains ~1400 rps with ~1400 connects/s (due to `-disable-keepalive`)

## Actual behavior

In networking monitoring on haproxy host I observe 1400/s active opens during the whole test, meaning haproxy does `connect` for each request

## Expected behavior

1400 active opens at start and ideally zero active opens per second during the rest of the test.

It was stated for 1.9: >Idle connections between HAProxy and the server are no longer closed immediately if the frontend connection vanishes. They remain open on the server to be used by other requests. Also the documentation for `pool-max-conn` states: >When idle
connections are enabled, orphaned idle connections which do not belong to any
client session anymore are moved to a dedicated pool so that they remain
usable by future clients.

**Comment by cognet** [comment]
Hi,

So despite what the doc says, and I'm actually about to change, sorry about that, setting pool-purge-delay to 0 actually disables the feature. Can you try with a higher pool-purge-delay ?
Also, if as a result of using -disable-keepalive you're sending a "Connection: close" header, I'm pretty sure we wont try to keep the server connection alive, if that happens, you may try your luck removing that header using "options http-use-htx" in your defaults section and "http-request del-header Connection" in your frontend section.

Thanks !

Olivier

**Comment by fillest** [comment]
Hi.
Ah, silly me! Indeed "Connection: close" should at least make backend close connections anyway.
I've applied your suggestions and the problems is gone. Thanks.

I guess I leave the issue opened for "pool-purge-delay 0" fix though.

**Comment by wtarreau** [comment]
The doc fix was merged in 1.9.6, closing.
